sdk/identity/src/token_credentials/default_credentials.rs

```diff
@@ -148,15 +248,19 @@ impl TokenCredential for DefaultAzureCredentialEnum {
 /// Consult the documentation of these credential types for more information on how they attempt authentication.
 #[derive(Debug)]
 pub struct DefaultAzureCredential {
-    sources: Vec<DefaultAzureCredentialEnum>,
+    sources: Vec<DefaultAzureCredentialKind>,
     cache: TokenCache,
 }
 
 impl DefaultAzureCredential {
+    pub fn create(options: TokenCredentialOptions) -> azure_core::Result<DefaultAzureCredential> {
+        DefaultAzureCredentialBuilder::default()
+            .with_options(options)
+            .build()
+    }
+
     /// Creates a `DefaultAzureCredential` with specified sources.
-    ///
-    /// These sources will be tried in the order provided in the `TokenCredential` authentication flow.
-    pub fn with_sources(sources: Vec<DefaultAzureCredentialEnum>) -> Self {
+    fn with_sources(sources: Vec<DefaultAzureCredentialKind>) -> Self {
```

Comment:
I removed `pub` here. It could be made public, but I'd like to know the use case.